pkg/backtest/matching.go

```diff
@@ -180,10 +188,20 @@ func (m *SimplePriceMatching) PlaceOrder(o types.SubmitOrder) (*types.Order, *ty
 	order := m.newOrder(o, orderID)
 
 	if isTaker {
+		var price fixedpoint.Value
 		if order.Type == types.OrderTypeMarket {
 			order.Price = m.Market.TruncatePrice(m.LastPrice)
+			price = order.Price
 		} else if order.Type == types.OrderTypeLimit {
-			order.AveragePrice = m.Market.TruncatePrice(m.LastPrice)
+			if m.NextKLine != nil && m.NextKLine.High.Compare(order.Price) > 0 && order.Side == types.SideTypeBuy {
```

Comment:
Can you add a comment describing this behavior?

Reply:
done